<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Siwei" />


<title>Crop-seq analysis by Siwei</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cropseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/cropseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Crop-seq analysis by Siwei</h1>
<h4 class="author"><em>Siwei</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-12-01</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20181119)</code> </summary></p>
<p>The command <code>set.seed(20181119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/simingz/cropseq/tree/fdd56472119b02d3dfe1fadd755df6b8b1db3682" target="_blank">fdd5647</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/Quality_metrics_cache/
    Ignored:    analysis/figure/

Unstaged changes:
    Modified:   analysis/Quality_metrics.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/simingz/cropseq/blob/fdd56472119b02d3dfe1fadd755df6b8b1db3682/analysis/crop_workflow_Siwei.Rmd" target="_blank">fdd5647</a>
</td>
<td style="text-align:left;">
szhao06
</td>
<td style="text-align:left;">
2018-12-01
</td>
<td style="text-align:left;">
qc
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/simingz/cropseq/fdd56472119b02d3dfe1fadd755df6b8b1db3682/docs/crop_workflow_Siwei.html" target="_blank">fdd5647</a>
</td>
<td style="text-align:left;">
szhao06
</td>
<td style="text-align:left;">
2018-12-01
</td>
<td style="text-align:left;">
qc
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/simingz/cropseq/blob/b5d096df68a9b0080889be7de234fae8926828a7/analysis/crop_workflow_Siwei.Rmd" target="_blank">b5d096d</a>
</td>
<td style="text-align:left;">
simingz
</td>
<td style="text-align:left;">
2018-11-29
</td>
<td style="text-align:left;">
migrate to rcc
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/simingz/cropseq/6360503650910fec56e1f698ca2d401c8b738dfe/docs/crop_workflow_Siwei.html" target="_blank">6360503</a>
</td>
<td style="text-align:left;">
simingz
</td>
<td style="text-align:left;">
2018-11-20
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/simingz/cropseq/blob/8be64d02528d178c3f9ed0934bcb75137b9f6e9d/analysis/crop_workflow_Siwei.Rmd" target="_blank">8be64d0</a>
</td>
<td style="text-align:left;">
simingz
</td>
<td style="text-align:left;">
2018-11-20
</td>
<td style="text-align:left;">
First
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="siweis-original-code" class="section level1">
<h1>Siwei’s original code</h1>
<pre class="r"><code># 01-Nov-2018
# obtain the gene expression profile of all gRNAs individually
library(edgeR)
library(Rfast)
library(data.table)
library(cellrangerRkit)
#################
# exp_matrix &lt;- exp_matrix_backup
##############
gbm &lt;- load_cellranger_matrix(&quot;../NSC_merged_05_07_08_new/&quot;) # load the GBM with NEW data 
exp_matrix &lt;- gbm[, colSums(exprs(gbm)) &gt; 11000] # get cells above background value
use_genes &lt;-get_nonzero_genes(exp_matrix)
exp_matrix &lt;- exp_matrix[use_genes, ]
exp_matrix &lt;- normalize_barcode_sums_to_median(exp_matrix)

# Select cells that have at least 1 reads in gRNA count
exp_matrix &lt;- exp_matrix[, colMaxs(as.matrix(exprs(exp_matrix[
  (length(exp_matrix@featureData@data$id)-75): # should be 75 or 76?
    (length(exp_matrix@featureData@data$id)), ])), value = T) &gt; 0] #more than 1 count

# make cell count for cells contain unique RNA
exp_matrix &lt;- exp_matrix[, # Select cells that have one gRNA expression more than 3x of all others
                         (colMaxs(as.matrix(exprs(exp_matrix[
                           (length(exp_matrix@featureData@data$id)-75):
                             (length(exp_matrix@featureData@data$id)), ])), value = T)) &gt;
                           (colSums(exprs(exp_matrix[(length(exp_matrix@featureData@data$id)-75):
                                                       (length(exp_matrix@featureData@data$id)), ]))*3/4)]

# exp_GeneBCMatrix &lt;- exp_matrix # output from the last step
#############

# find the gRNA for each cell by taking the rowMax index of gRNA columns(29847:29922)
# will return a data.frame with entries followed by .xx since colnames should be unique
exp_matrix &lt;- as.matrix(exprs(exp_matrix))
exp_matrix_backup &lt;- exp_matrix # make a backup in case of need

#############
exp_matrix &lt;- exp_matrix_backup
############
# total UMI count: 77,997,203
#scale exp_matrix to 1,000,000 counts -- NO, will significantly decrease the statistic power
# exp_matrix &lt;- exp_matrix*(1000000/77997203)


### Make the gRNA list
# gRNA_list &lt;- as.data.frame(rownames(exp_matrix
#                                     [(nrow(exp_matrix)-76 +
#                                         colMaxs(
#                                           as.matrix(
#                                             exp_matrix[
#                                               ((nrow(exp_matrix)-75):(nrow(exp_matrix)))
#                                               ,]                                             
#                                           ), value = F)
#                                     ),
#                                       ]))
gRNA_list &lt;- as.data.frame(rownames(exp_matrix
                                    [(nrow(exp_matrix)-76 +
                                        colMaxs(
                                            exp_matrix[
                                              ((nrow(exp_matrix)-75):(nrow(exp_matrix)))
                                              ,]                                             
                                          , value = F)
                                    ),
                                      ]))

colnames(gRNA_list) &lt;- &quot;gRNA&quot;

gRNA_ASoC_list &lt;- gRNA_list[!duplicated(gRNA_list$gRNA), ]
gRNA_ASoC_list &lt;- gRNA_ASoC_list[order(gRNA_ASoC_list)]
# exp_matrix$cell_type &lt;- as.factor(gRNA_dist$gRNA)
gRNA_ASoC_list &lt;- gRNA_ASoC_list[-c(31,32,33,37,38,39,40,41,51,52,59,60,61,65,66,67)]
#############
# exp_matrix now: rownames=ENSG, colnames=barcode


# exp_matrix &lt;- as.matrix(t(exp_matrix)) 
# not using
cell_type_index &lt;- as.vector(gRNA_list$gRNA) # use this variable as index, length = 2522

############ Main Program ######
gRNA_ASoC_list_backup &lt;- gRNA_ASoC_list # set a backup, will use the full list after debugging

# gRNA_ASoC_list &lt;- gRNA_ASoC_list[1:2] # list for debugging, 2 elements only


# gRNA_ASoC_list &lt;- gRNA_ASoC_list_backup

for(i in 1:length(gRNA_ASoC_list)) {
  print(gRNA_ASoC_list[i])
  ### make correspondence of Gene_Symbol and Gene_id
  # write_out_with_gene_name &lt;- as.data.frame(get_CRISPRi_result(gRNA_ASoC_list[i], 
  #                                                              TPM_filter = F, TPM_threshold = 0.01))
  write_out_with_gene_name &lt;- as.data.frame(get_CRISPRi_result(gRNA_ASoC_list[i], 
                                                               TPM_filter = F, TPM_threshold = 0.01)) # no filter set
  write_out_with_gene_name$Geneid &lt;- rownames(write_out_with_gene_name)
  write_out_with_gene_name_output &lt;- merge(write_out_with_gene_name, ENSG_coord_gene_gencodev28, by = &quot;Geneid&quot;)
  write_out_with_gene_name_output &lt;- write_out_with_gene_name_output[order(write_out_with_gene_name_output$PValue), ]
  write.table(write_out_with_gene_name_output, append = F, 
              row.names = F, col.names = T, sep = &quot;\t&quot;, quote = F, 
              file = paste(&quot;output/&quot;, gRNA_ASoC_list[i], &quot;_gRNA.txt&quot;, collapse = &quot;&quot;, sep = &quot;&quot;))
}

############

############## custom functions##############

get_CRISPRi_result &lt;- function(gRNA_name, TPM_filter = FALSE, TPM_threshold = 0.01) {

    # TPM filter takes only genes with an estimated TPM above 1
  # in more than 25% of the considered cells
  # TPM_filter &lt;- TRUE # temporary
  # TPM_threshold &lt;- 0.01 # temporaray
  # if(TPM_filter) {
  #   exp_matrix &lt;- exp_matrix[ 
  #     rowSums(exp_matrix &gt; TPM_threshold) 
  #     &gt; trunc(ncol(exp_matrix/4)), ]
  # }
  # make two matrix using grepl, separate the target gRNA and control gRNA (EGFP/neg)
  control_matrix &lt;- exp_matrix[ , grepl(&quot;EGFP&quot;, cell_type_index) |
                                 grepl(&quot;neg&quot;, cell_type_index)]
  # gRNA_name &lt;- &quot;VPS45_2_gene&quot; # temporary
  gRNA_matrix &lt;- exp_matrix[ , grepl(gRNA_name, cell_type_index)]
 
  # prepare the input matrix for edgeR
  ## merge the two matrices as one, ensure the gRNA is the first
  matrix_combined &lt;- as.matrix(cbind(control_matrix, gRNA_matrix))

  # matrix_combined &lt;- matrix_combined[order(cell_type_index), ]
  ## Trim the tailing gRNA artificial genes, total 75
  # matrix_combined_transposed &lt;- t(matrix_combined[, -((ncol(matrix_combined)-75):
  #                                                       ncol(matrix_combined))])
  matrix_combined_transposed &lt;- matrix_combined[1:(nrow(matrix_combined)-76), ]
  ### Assign rownames as ENSG gene identifiers
  rownames(matrix_combined_transposed) &lt;- gsub(&quot;\\..*&quot;, &quot;&quot;, rownames(matrix_combined_transposed))
  # rownames(matrix_combined_transposed) &lt;- gsub(&quot;\\..*&quot;, &quot;&quot;, 
  #                                              colnames(matrix_combined[ , 1:(ncol(matrix_combined)-76)]))
  # colnames(matrix_combined_transposed) &lt;- rownames(matrix_combined)

  # print(scale(colMeans(matrix_combined_transposed &gt; 0)))
  # Run edgeR use edgeRQLFDetRate, nrow(control_matrix) should be 139
  group &lt;- factor(c(rep(&quot;ctrl&quot;, len = ncol(control_matrix)), 
                    rep(&quot;gRNA&quot;, len = ncol(gRNA_matrix))))
  # make DGEList()
  main_DGE &lt;- DGEList(counts = matrix_combined_transposed, group = group, remove.zeros = T)
  # Use edgeRQLFDetRate flow from now on
  main_DGE &lt;- calcNormFactors(main_DGE)
  cdr &lt;- scale(colMeans(matrix_combined_transposed &gt; 0)) # DetRate is applied here
  design &lt;- model.matrix(~ cdr + group) # cdr (~ cdr + group)
  main_DGE &lt;- estimateDisp(main_DGE, design = design)
  fit &lt;- glmQLFit(main_DGE, design = design) # fit
  qlf &lt;- glmQLFTest(fit) # QLF vs LRT
  # tt &lt;- topTags(qlf, n = 100)
  exp_table &lt;- qlf$table
  exp_table$FDR &lt;- p.adjust(exp_table$PValue, &quot;fdr&quot;)
  
  # write.table(group, append = F, 
  #             row.names = F, col.names = T, sep = &quot;\t&quot;, quote = F, 
  #             file = paste(&quot;output/&quot;, gRNA_ASoC_list[i], &quot;_group.txt&quot;, collapse = &quot;&quot;))
  
  return(exp_table)
}</code></pre>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin11.4.2 (64-bit)
Running under: OS X El Capitan 10.11.6

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.1.1   Rcpp_0.12.17      digest_0.6.12    
 [4] rprojroot_1.2     R.methodsS3_1.7.1 backports_1.0.5  
 [7] git2r_0.18.0      magrittr_1.5      evaluate_0.10    
[10] stringi_1.1.5     whisker_0.3-2     R.oo_1.22.0      
[13] R.utils_2.7.0     rmarkdown_1.10    tools_3.3.2      
[16] stringr_1.2.0     yaml_2.1.16       htmltools_0.3.6  
[19] knitr_1.20       </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
