<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Siming Zhao" />


<title>Differential expression analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cropseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/cropseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis</h1>
<h4 class="author"><em>Siming Zhao</em></h4>
<h4 class="date"><em>December 2, 2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-12-05</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20181119)</code> </summary></p>
<p>The command <code>set.seed(20181119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/simingz/cropseq/tree/275d5d865e08b6de5e4c9d3d038522c52b998039" target="_blank">275d5d8</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Quality_metrics_cache/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/DE_analysis.Rmd
    Untracked:  code/qq-plot.R
    Untracked:  code/summary_functions.R
    Untracked:  docs/figure/DE_analysis.Rmd/

Unstaged changes:
    Modified:   analysis/Quality_metrics.Rmd
    Modified:   analysis/crop_workflow_Alan.Rmd
    Modified:   analysis/index.Rmd
    Modified:   data/DE_input.Rd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<hr />
<p>Load data</p>
<pre class="r"><code>source(&quot;code/summary_functions.R&quot;)
load(&quot;data/DE_input.Rd&quot;)
glocus &lt;- &quot;VPS45&quot;
dim(dm)[1]</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>gcount &lt;- dm[1:(dim(dm)[1]-76), colnames(dm1dfagg)[dm1dfagg[glocus,] &gt;0 &amp; nlocus==1]]
# negative control cells defined as neg gRNA targeted cells
ncount &lt;- dm[1:(dim(dm)[1]-76), colnames(dm1dfagg)[dm1dfagg[&quot;neg&quot;,] &gt;0 &amp; nlocus==1]]
coldata &lt;- data.frame(row.names = c(colnames(gcount),colnames(ncount)),     
                      condition=c(rep(&#39;G&#39;,dim(gcount)[2]),rep(&#39;N&#39;,dim(ncount)[2])))</code></pre>
<div id="deseq2" class="section level2">
<h2>DEseq2</h2>
<div id="standard-deseq2" class="section level3">
<h3>standard DESeq2</h3>
<pre class="r"><code>library(DESeq2)
dds = DESeqDataSetFromMatrix(countData = cbind(gcount,ncount),
                             colData = coldata,
                             design = ~condition)

dds = estimateSizeFactors(dds)
ddsWARD = DESeq(dds)
resWARD = results(ddsWARD)
summ_pvalues(resWARD$pvalue[!is.na(resWARD$pvalue)])</code></pre>
<p><img src="figure/DE_analysis.Rmd/deseq2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resSigWARD &lt;- subset(resWARD, padj &lt; 0.1)</code></pre>
<p>There are 0 genes passed FDR &lt;0.1 cutoff.</p>
</div>
<div id="deseq2-with-lrt-test" class="section level3">
<h3>DESeq2 with LRT test</h3>
<p>Following recommendation for single cell from <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#recommendations-for-single-cell-analysis">here</a>.</p>
<pre class="r"><code>ddsLRT = DESeq(dds, test=&quot;LRT&quot;, reduced = ~1,  sfType=&quot;poscounts&quot;, useT=TRUE, minmu=1e-6,minReplicatesForReplace=Inf)
resLRT = results(ddsLRT)
summ_pvalues(resLRT$pvalue[!is.na(resLRT$pvalue)])</code></pre>
<p><img src="figure/DE_analysis.Rmd/deseq2-lrt-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resSigLRT &lt;- subset(resLRT, padj &lt; 0.1)</code></pre>
<p>There are 0 genes passed FDR &lt;0.1 cutoff.</p>
</div>
</div>
<div id="edger" class="section level2">
<h2>edgeR</h2>
<div id="quasi-likelihood-f-tests" class="section level3">
<h3>quasi-likelihood F-tests</h3>
<pre class="r"><code>library(edgeR)
y &lt;- DGEList(counts= cbind(gcount,ncount),group=coldata$condition)
y &lt;- calcNormFactors(y)
group=coldata$condition
design &lt;- model.matrix(~group)
y &lt;- estimateDisp(y,design)
fitqlf &lt;- glmQLFit(y,design)
qlf &lt;- glmQLFTest(fitqlf,coef=2)
summ_pvalues(qlf$table$PValue)</code></pre>
<p><img src="figure/DE_analysis.Rmd/edgeR-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>topTags(qlf)</code></pre>
<pre><code>Coefficient:  groupN 
                       logFC   logCPM        F       PValue          FDR
ENSG00000176956.12 -2.776350 6.601479 58.72481 3.918884e-13 1.169630e-08
ENSG00000100097.11 -2.299929 6.624158 45.44607 1.064435e-10 1.171374e-06
ENSG00000130203.9  -1.831334 6.391155 45.21290 1.177418e-10 1.171374e-06
ENSG00000100300.17 -1.607537 6.410639 40.80264 8.073546e-10 6.024077e-06
ENSG00000138136.6  -1.976572 6.423656 39.44806 1.468212e-09 8.764051e-06
ENSG00000089116.3  -1.542926 6.320288 36.99506 1.605330e-08 7.578892e-05
ENSG00000175899.14 -1.584250 6.857262 33.87858 1.777533e-08 7.578892e-05
ENSG00000162992.3  -1.608906 6.373881 34.49478 2.385458e-08 8.899547e-05
ENSG00000198417.6  -1.600763 6.403261 32.30436 3.635396e-08 1.205578e-04
ENSG00000104327.7  -1.374926 6.326720 49.50835 7.206623e-08 2.150889e-04</code></pre>
</div>
<div id="likelihood-ratio-tests" class="section level3">
<h3>likelihood ratio tests</h3>
<pre class="r"><code>fitlrt &lt;- glmFit(y,design)
lrt &lt;- glmLRT(fitlrt,coef=2)
topTags(lrt)</code></pre>
<pre><code>Coefficient:  groupN 
                        logFC   logCPM       LR       PValue       FDR
ENSG00000175899.14 -1.4640348 6.857262 19.83212 8.454987e-06 0.1170441
ENSG00000176956.12 -2.6064930 6.601479 19.25383 1.144407e-05 0.1170441
ENSG00000100097.11 -2.2150044 6.624158 19.20106 1.176480e-05 0.1170441
ENSG00000100300.17 -1.4959097 6.410639 14.35365 1.514859e-04 0.9651347
ENSG00000119906.11  1.0518706 6.444915 13.90103 1.926926e-04 0.9651347
ENSG00000185900.9  -0.7876253 6.269218 13.68055 2.166870e-04 0.9651347
ENSG00000219626.8  -0.9907215 6.458688 13.59855 2.263601e-04 0.9651347
ENSG00000078061.12 -0.8304381 6.581079 12.89330 3.297606e-04 1.0000000
ENSG00000234912.11  1.0422011 6.412729 12.49191 4.087177e-04 1.0000000
ENSG00000175806.14 -0.7904605 6.612756 12.19362 4.795325e-04 1.0000000</code></pre>
<pre class="r"><code>summ_pvalues(lrt$table$PValue)</code></pre>
<p><img src="figure/DE_analysis.Rmd/edgeR-lrt-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="quasi-likelihood-f-tests-with-prefiltering" class="section level3">
<h3>quasi-likelihood F-tests with prefiltering</h3>
<p>Filter genes with 0 coverage in all cells.</p>
<pre class="r"><code>mycount &lt;- cbind(gcount,ncount)
totalcount &lt;- apply(mycount,1,sum)
y &lt;- DGEList(counts= mycount[totalcount&gt;0,],group=coldata$condition)
y &lt;- calcNormFactors(y)
group=coldata$condition
design &lt;- model.matrix(~group)
y &lt;- estimateDisp(y,design)
fitqlf &lt;- glmQLFit(y,design)
qlf &lt;- glmQLFTest(fitqlf,coef=2)
topTags(qlf)</code></pre>
<pre><code>Coefficient:  groupN 
                       logFC   logCPM        F       PValue          FDR
ENSG00000176956.12 -2.776350 6.601479 58.72481 3.918884e-13 6.170676e-09
ENSG00000100097.11 -2.299929 6.624158 45.44607 1.064435e-10 6.179874e-07
ENSG00000130203.9  -1.831334 6.391155 45.21290 1.177418e-10 6.179874e-07
ENSG00000100300.17 -1.607537 6.410639 40.80264 8.073546e-10 3.178152e-06
ENSG00000138136.6  -1.976572 6.423656 39.44806 1.468212e-09 4.623693e-06
ENSG00000089116.3  -1.542926 6.320288 36.99506 1.605330e-08 3.998433e-05
ENSG00000175899.14 -1.584250 6.857262 33.87858 1.777533e-08 3.998433e-05
ENSG00000162992.3  -1.608906 6.373881 34.49478 2.385458e-08 4.695177e-05
ENSG00000198417.6  -1.600763 6.403261 32.30436 3.635396e-08 6.360327e-05
ENSG00000104327.7  -1.374926 6.326720 49.50835 7.206623e-08 1.134755e-04</code></pre>
<pre class="r"><code>summ_pvalues(qlf$table$PValue)</code></pre>
<p><img src="figure/DE_analysis.Rmd/edgeR-flt-filter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="likelihood-ratio-tests-with-prefiltering" class="section level3">
<h3>likelihood ratio tests with prefiltering</h3>
<p>Filter genes with 0 coverage in all cells.</p>
<pre class="r"><code>fitlrt &lt;- glmFit(y,design)
lrt &lt;- glmLRT(fitlrt,coef=2)
topTags(lrt)</code></pre>
<pre><code>Coefficient:  groupN 
                        logFC   logCPM       LR       PValue        FDR
ENSG00000175899.14 -1.4640348 6.857262 19.83212 8.454987e-06 0.06174953
ENSG00000176956.12 -2.6064930 6.601479 19.25383 1.144407e-05 0.06174953
ENSG00000100097.11 -2.2150044 6.624158 19.20106 1.176480e-05 0.06174953
ENSG00000100300.17 -1.4959097 6.410639 14.35365 1.514859e-04 0.50918085
ENSG00000119906.11  1.0518706 6.444915 13.90103 1.926926e-04 0.50918085
ENSG00000185900.9  -0.7876253 6.269218 13.68055 2.166870e-04 0.50918085
ENSG00000219626.8  -0.9907215 6.458688 13.59855 2.263601e-04 0.50918085
ENSG00000078061.12 -0.8304381 6.581079 12.89330 3.297606e-04 0.64905134
ENSG00000234912.11  1.0422011 6.412729 12.49191 4.087177e-04 0.66841632
ENSG00000175806.14 -0.7904605 6.612756 12.19362 4.795325e-04 0.66841632</code></pre>
<pre class="r"><code>summ_pvalues(lrt$table$PValue)</code></pre>
<p><img src="figure/DE_analysis.Rmd/edgeR-lrt-filter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="parameters-used" class="section level2">
<h2>Parameters used</h2>
<ul>
<li>We used data processed after QC step <a href="Quality_metrics.html">here</a>.<br />
</li>
<li>EdgeR: Prefiltering of lowly expressed genes: genes with 0 coverage in all cells. DEseq2 suggests that this pre-filtering step is only useful to increase speed, not for multiple testing purposes, so no filtering.</li>
<li>targeted locus, choose VPS45.</li>
</ul>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin14.5.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] edgeR_3.22.5                limma_3.36.5               
 [3] gridExtra_2.3               lattice_0.20-35            
 [5] DESeq2_1.20.0               SummarizedExperiment_1.10.1
 [7] DelayedArray_0.6.6          BiocParallel_1.14.2        
 [9] matrixStats_0.54.0          Biobase_2.40.0             
[11] GenomicRanges_1.32.7        GenomeInfoDb_1.16.0        
[13] IRanges_2.14.12             S4Vectors_0.18.3           
[15] BiocGenerics_0.26.0         Matrix_1.2-14              

loaded via a namespace (and not attached):
 [1] bit64_0.9-7            splines_3.5.1          R.utils_2.7.0         
 [4] Formula_1.2-3          assertthat_0.2.0       latticeExtra_0.6-28   
 [7] blob_1.1.1             GenomeInfoDbData_1.1.0 yaml_2.2.0            
[10] RSQLite_2.1.1          pillar_1.3.0           backports_1.1.2       
[13] glue_1.3.0             digest_0.6.18          RColorBrewer_1.1-2    
[16] XVector_0.20.0         checkmate_1.8.5        colorspace_1.3-2      
[19] htmltools_0.3.6        R.oo_1.22.0            plyr_1.8.4            
[22] XML_3.98-1.16          pkgconfig_2.0.2        genefilter_1.62.0     
[25] zlibbioc_1.26.0        purrr_0.2.5            xtable_1.8-3          
[28] scales_1.0.0           whisker_0.3-2          git2r_0.23.0          
[31] tibble_1.4.2           htmlTable_1.12         annotate_1.58.0       
[34] ggplot2_3.1.0          nnet_7.3-12            lazyeval_0.2.1        
[37] survival_2.42-6        magrittr_1.5           crayon_1.3.4          
[40] memoise_1.1.0          evaluate_0.12          R.methodsS3_1.7.1     
[43] foreign_0.8-71         tools_3.5.1            data.table_1.11.6     
[46] stringr_1.3.1          locfit_1.5-9.1         munsell_0.5.0         
[49] cluster_2.0.7-1        AnnotationDbi_1.42.1   bindrcpp_0.2.2        
[52] compiler_3.5.1         rlang_0.3.0.1          RCurl_1.95-4.11       
[55] rstudioapi_0.8         htmlwidgets_1.2        bitops_1.0-6          
[58] base64enc_0.1-3        rmarkdown_1.10         gtable_0.2.0          
[61] DBI_1.0.0              R6_2.3.0               knitr_1.20            
[64] dplyr_0.7.6            bit_1.1-12             bindr_0.1.1           
[67] Hmisc_4.1-1            workflowr_1.1.1        rprojroot_1.3-2       
[70] stringi_1.2.4          Rcpp_1.0.0             geneplotter_1.58.0    
[73] rpart_4.1-13           acepack_1.4.1          tidyselect_0.2.4      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
