<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Siming Zhao" />


<title>Quality_metrics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cropseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/simingz/cropseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Quality_metrics</h1>
<h4 class="author"><em>Siming Zhao</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cropseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
Alanmatrix
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>Quality_metrics_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsimingzcropseqtree8554c826aa0ffc8b284d1f4ba814c1f3afb47a23targetblank8554c82a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/simingz/cropseq/tree/8554c826aa0ffc8b284d1f4ba814c1f3afb47a23" target="_blank">8554c82</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsimingzcropseqtree8554c826aa0ffc8b284d1f4ba814c1f3afb47a23targetblank8554c82a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/Quality_metrics_cache/
    Ignored:    analysis/figure/gRNA-EdgeR-QLF.Rmd/
    Ignored:    analysis/figure/gRNA-EdgeR-QLF_permutation.Rmd/
    Ignored:    data/gRNA_edgeR-QLF/
    Ignored:    data/gRNA_edgeR-QLF_811d97b/
    Ignored:    data/gRNA_edgeR-QLF_fba9768/

Untracked files:
    Untracked:  analysis/MASTcpmdet.Rmd

Unstaged changes:
    Modified:   analysis/Quality_metrics.Rmd
    Modified:   code/DE_functions.R
    Modified:   code/WIP_2019.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/analysis/Quality_metrics.Rmd" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
<td>
permutation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/01a5914362313fd7fbf78c601924e29e58314200/docs/Quality_metrics.html" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
<td>
permutation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/49ecf6e2cf5686b9d11690f78c565c7361d08d6e/analysis/Quality_metrics.Rmd" target="_blank">49ecf6e</a>
</td>
<td>
simingz
</td>
<td>
2018-12-16
</td>
<td>
explore filtering
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/6b6ebde01577aff334bc59802022d3cb6c81a272/analysis/Quality_metrics.Rmd" target="_blank">6b6ebde</a>
</td>
<td>
simingz
</td>
<td>
2018-12-14
</td>
<td>
fix Xin’s comment for qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/6b6ebde01577aff334bc59802022d3cb6c81a272/docs/Quality_metrics.html" target="_blank">6b6ebde</a>
</td>
<td>
simingz
</td>
<td>
2018-12-14
</td>
<td>
fix Xin’s comment for qc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/8ce79ed3ec7d44f3ad745fd88b4e9eab6affcd0a/analysis/Quality_metrics.Rmd" target="_blank">8ce79ed</a>
</td>
<td>
simingz
</td>
<td>
2018-12-05
</td>
<td>
de_anlysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/8ce79ed3ec7d44f3ad745fd88b4e9eab6affcd0a/docs/Quality_metrics.html" target="_blank">8ce79ed</a>
</td>
<td>
simingz
</td>
<td>
2018-12-05
</td>
<td>
de_anlysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/275d5d865e08b6de5e4c9d3d038522c52b998039/analysis/Quality_metrics.Rmd" target="_blank">275d5d8</a>
</td>
<td>
simingz
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/275d5d865e08b6de5e4c9d3d038522c52b998039/docs/Quality_metrics.html" target="_blank">275d5d8</a>
</td>
<td>
simingz
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/f5dda86f5471ff786625c5809a81d7e984730d04/analysis/Quality_metrics.Rmd" target="_blank">f5dda86</a>
</td>
<td>
simingz
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/f5dda86f5471ff786625c5809a81d7e984730d04/docs/Quality_metrics.html" target="_blank">f5dda86</a>
</td>
<td>
simingz
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/8754cad65d01c5766bac60611204c4cdd22bb23a/analysis/Quality_metrics.Rmd" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/8754cad65d01c5766bac60611204c4cdd22bb23a/docs/Quality_metrics.html" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/c206c9d2a24d343aaf9dae5f78737a21d719db50/analysis/Quality_metrics.Rmd" target="_blank">c206c9d</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-01
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/simingz/cropseq/blob/fdd56472119b02d3dfe1fadd755df6b8b1db3682/analysis/Quality_metrics.Rmd" target="_blank">fdd5647</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-01
</td>
<td>
qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/simingz/cropseq/fdd56472119b02d3dfe1fadd755df6b8b1db3682/docs/Quality_metrics.html" target="_blank">fdd5647</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-01
</td>
<td>
qc
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="number-of-guide-rnas-per-cell" class="section level2">
<h2>Number of guide RNAs per cell</h2>
<ul>
<li>number of cells with guide RNA reads =1 From Siwei’s cellranger run:</li>
</ul>
<pre class="r"><code>library(Matrix)
matrix_dir = &quot;/project2/xinhe/simingz/CROP-seq/data_from_Siwei/Xin_scRNA_seq_05Nov2018/filtered_gene_bc_matrices/CellRanger_index/&quot;
matrix.path &lt;- paste0(matrix_dir, &quot;matrix.mtx&quot;)
dm &lt;- readMM(file = matrix.path)
dm1 &lt;- tail(dm,n=76)
length(colSums(dm1)[colSums(dm1)==1])</code></pre>
<pre><code>[1] 440</code></pre>
<p>From Alan’s cellranger run:</p>
<pre class="r"><code>matrix_dir1 = &quot;/project2/xinhe/simingz/CROP-seq/NSC0507_cellranger/outs/filtered_gene_bc_matrices/cellranger_ref/&quot;
matrix.path1 &lt;- paste0(matrix_dir1, &quot;matrix.mtx&quot;)
mattemp1 &lt;- readMM(file = matrix.path1)
mattemp11 &lt;- tail(mattemp1,n=76)
length(colSums(mattemp11)[colSums(mattemp11)==1])</code></pre>
<pre><code>[1] 266</code></pre>
<pre class="r"><code>matrix_dir2 = &quot;/project2/xinhe/simingz/CROP-seq/NSC08_cellranger/outs/filtered_gene_bc_matrices/cellranger_ref/&quot;
matrix.path2 &lt;- paste0(matrix_dir2, &quot;matrix.mtx&quot;)
mattemp2 &lt;- readMM(file = matrix.path2)
mattemp21 &lt;- tail(mattemp2,n=76)
length(colSums(mattemp21)[colSums(mattemp21)==1])</code></pre>
<pre><code>[1] 190</code></pre>
<p>Note: in Alan’s original analysis conversion from h5 to csv step didn’t seem to work properly. if starting from matrix.mtx files. Siwei and Alan’s analyses gave the same results. So from now on, we will always start from Siwei’s matrix.mtx file.</p>
<ul>
<li>distribution of gRNA types per cell</li>
</ul>
<pre class="r"><code>barcode.path &lt;- paste0(matrix_dir, &quot;barcodes.tsv&quot;)
features.path &lt;- paste0(matrix_dir, &quot;genes.tsv&quot;)
feature.names = read.delim(features.path, header = FALSE,
                           stringsAsFactors = FALSE)
barcode.names = read.delim(barcode.path, header = FALSE,
                           stringsAsFactors = FALSE)
colnames(dm) = barcode.names$V1
rownames(dm) = feature.names$V2
dm1 &lt;- tail(dm,n=76)

hist(apply(dm1, 2, function(x) length(x[x&gt;0])),breaks=300,xlim=c(0,15),ylim=c(0,2500), main=&quot;Distribution of number of gRNA types per cell&quot;, xlab= &quot;# gRNA type per cell&quot;)</code></pre>
<p><img src="figure/Quality_metrics.Rmd/gRNAdist-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gRNAdist-1">
Past versions of gRNAdist-1.png
</button>
</p>
<div id="fig-gRNAdist-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/gRNAdist-1.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/6b6ebde01577aff334bc59802022d3cb6c81a272/docs/figure/Quality_metrics.Rmd/gRNAdist-1.png" target="_blank">6b6ebde</a>
</td>
<td>
simingz
</td>
<td>
2018-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/fdd56472119b02d3dfe1fadd755df6b8b1db3682/docs/figure/Quality_metrics.Rmd/gRNAdist-1.png" target="_blank">fdd5647</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="number-of-cells-targeted-for-each-locus" class="section level2">
<h2>number of cells targeted for each locus</h2>
<pre class="r"><code>library(dplyr)
dm1df &lt;- as.data.frame(as.matrix(dm1))
dm1df$label = sapply(strsplit(rownames(dm1),split = &#39;_&#39;), function(x){x[1]})
dm1dfagg = as.data.frame(dm1df %&gt;% group_by(label) %&gt;% summarise_all(funs(sum)))
row.names(dm1dfagg) =dm1dfagg$label
dm1dfagg$label =NULL</code></pre>
<ul>
<li>number of cells targeted for each locus</li>
</ul>
<pre class="r"><code>ncell &lt;- apply(dm1dfagg,1, function (x) length(x[x&gt;=1]))
barplot(ncell,las=2,cex.lab=1, main= &quot;# cells targted for each locus&quot;)</code></pre>
<p><img src="figure/Quality_metrics.Rmd/celldist1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-celldist1-1">
Past versions of celldist1-1.png
</button>
</p>
<div id="fig-celldist1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/celldist1-1.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/8754cad65d01c5766bac60611204c4cdd22bb23a/docs/figure/Quality_metrics.Rmd/celldist1-1.png" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/fdd56472119b02d3dfe1fadd755df6b8b1db3682/docs/figure/Quality_metrics.Rmd/celldist1-1.png" target="_blank">fdd5647</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>number of cells only targeted for that locus</li>
</ul>
<pre class="r"><code># Singletons (cells with only 1 gRNA)
nlocus &lt;- apply(dm1dfagg, 2, function (x) length(x[x&gt;=1]))
hist(nlocus,breaks=100, main=&quot;number of targeted locus each cell&quot;)</code></pre>
<p><img src="figure/Quality_metrics.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/unnamed-chunk-1-1.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/8754cad65d01c5766bac60611204c4cdd22bb23a/docs/figure/Quality_metrics.Rmd/unnamed-chunk-1-1.png" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dm1dfagg.uni= dm1dfagg[,nlocus==1]

ncell.uni &lt;- apply(dm1dfagg.uni,1, function (x) length(x[x&gt;=1]))
barplot(ncell.uni,las=2,cex.lab=1,main= &quot;# cells uniquely targted for each locus&quot;)</code></pre>
<p><img src="figure/Quality_metrics.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/unnamed-chunk-1-2.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/8754cad65d01c5766bac60611204c4cdd22bb23a/docs/figure/Quality_metrics.Rmd/unnamed-chunk-1-2.png" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="umi-count-distribution-for-cells-with-unique-targeted-locus" class="section level2">
<h2>UMI count distribution for cells with unique targeted locus</h2>
<pre class="r"><code># Singletons (cells with only 1 targeted locus)
dm.uni &lt;- dm[,nlocus==1]
nUMI &lt;- colSums(dm.uni)
hist(nUMI,breaks=100,xlim=c(0,1e5))  </code></pre>
<p><img src="figure/Quality_metrics.Rmd/UMI-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-UMI-1">
Past versions of UMI-1.png
</button>
</p>
<div id="fig-UMI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/UMI-1.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/8754cad65d01c5766bac60611204c4cdd22bb23a/docs/figure/Quality_metrics.Rmd/UMI-1.png" target="_blank">8754cad</a>
</td>
<td>
szhao06
</td>
<td>
2018-12-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="umi-count-distribution-for-grnas-in-cells-with-unique-targeted-locus" class="section level2">
<h2>UMI count distribution for gRNAs in cells with unique targeted locus</h2>
<pre class="r"><code># Singletons (cells with only 1 targeted locus)
nUMIgRNA &lt;- colSums(tail(dm.uni,76))
hist(nUMIgRNA,breaks=500,xlim=c(0,20), main = &quot;Histogram of nUMI for gRNAs&quot;)  </code></pre>
<p><img src="figure/Quality_metrics.Rmd/UMIgRNA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-UMIgRNA-1">
Past versions of UMIgRNA-1.png
</button>
</p>
<div id="fig-UMIgRNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/01a5914362313fd7fbf78c601924e29e58314200/docs/figure/Quality_metrics.Rmd/UMIgRNA-1.png" target="_blank">01a5914</a>
</td>
<td>
simingz
</td>
<td>
2019-02-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/simingz/cropseq/blob/6b6ebde01577aff334bc59802022d3cb6c81a272/docs/figure/Quality_metrics.Rmd/UMIgRNA-1.png" target="_blank">6b6ebde</a>
</td>
<td>
simingz
</td>
<td>
2018-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="prepare-data-for-differential-gene-expression" class="section level2">
<h2>Prepare data for differential gene expression</h2>
<p>Rows with duplicated gene names will be removed</p>
<pre class="r"><code>table(rownames(dm))[table(rownames(dm))&gt;1]</code></pre>
<pre><code>
  AJ271736.10       AKAP17A          ASMT         ASMTL          CD99 
            2             2             2             2             2 
        CRLF2        CSF2RA         DHRSX        GTPBP6         IL3RA 
            2             2             2             2             2 
         IL9R       KLHDC7B       MIR1253     MIR3179-1     MIR3179-3 
            2             2             2             2             2 
    MIR3180-1     MIR3180-2     MIR3180-3     MIR3180-4       MIR3690 
            2             2             2             2             2 
      MIR6089    NCRNA00102    NCRNA00106         P2RY8        PLCXD1 
            2             2             2             2             2 
      PPP2R3B RP11-309M23.1 RP13-297E16.4 RP13-297E16.5 RP13-465B17.5 
            2             2             2             2             2 
         SHOX       SLC25A6         SPRY3         VAMP7         ZBED1 
            2             2             2             2             2 </code></pre>
<pre class="r"><code> dm &lt;- dm[!(rownames(dm) %in% names(table(rownames(dm))[table(rownames(dm))&gt;1])), ]
save(dm,dm1dfagg,nlocus, file=&quot;data/DE_input.Rd&quot;)</code></pre>
<p>Parameters used:</p>
<ul>
<li><p>for a cell to be considered targeted uniquely at a locus: total read counts for the 3 gRNAs targeting that locus &gt;1, total read counts for gRNA of other locus=0.</p></li>
<li><p>negative control: neg_EGFP and neg_CTRL are pooled together.</p></li>
<li><p>cells to be exluded due to low total UMI count: no filtering</p></li>
</ul>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.7.8   Matrix_1.2-15

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0       knitr_1.20       bindr_0.1.1      whisker_0.3-2   
 [5] magrittr_1.5     workflowr_1.2.0  tidyselect_0.2.5 lattice_0.20-38 
 [9] R6_2.3.0         rlang_0.3.1      stringr_1.4.0    tools_3.5.1     
[13] grid_3.5.1       git2r_0.23.0     htmltools_0.3.6  yaml_2.2.0      
[17] rprojroot_1.3-2  digest_0.6.18    assertthat_0.2.0 tibble_2.0.1    
[21] crayon_1.3.4     bindrcpp_0.2.2   purrr_0.3.2      fs_1.2.6        
[25] glue_1.3.0       evaluate_0.12    rmarkdown_1.10   stringi_1.3.1   
[29] pillar_1.3.1     compiler_3.5.1   backports_1.1.2  pkgconfig_2.0.2 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
