---
title: "cis gene enrichment"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Get cis genes for each targted locus
```{bash, eval=F}
# SNPfile: "/project2/xinhe/simingz/CROP-seq/scRNA_seq_SNP_list.txt"
module load mysql
mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -D hg19 -e '
select
 K.name2,
 K.name,
 S.name,
 S.avHet,
 S.chrom,
 S.chromStart,
 K.txStart,
 K.txEnd
from snp150 as S
left join refGene as K on
 (S.chrom=K.chrom and not(K.txEnd+50000<S.chromStart or S.chromEnd+50000<K.txStart))
where
 S.name in ("rs7148456","rs12895055","rs7170068","rs520843","rs12716973","rs2192932","rs17200916","rs1198588","rs324017","rs4151680","rs301791","rs324015","rs9882911","rs11633075","rs2027349","rs186132169","rs9661794","rs7936858","rs3861678","rs10933","rs6071578")' > /project2/xinhe/simingz/CROP-seq/cropseq/data/SNP_50000.txt
```

## genes +/- 50kb of targeted SNPs
```{r, gRNAsnp}
snpfile <- "/project2/xinhe/simingz/CROP-seq/scRNA_seq_SNP_list.txt"
gRNAsnp <- read.table(snpfile, header=F, stringsAsFactors = F)
colnames(gRNAsnp) <- c("locus_SNP", "locus")
```

```{r, er50k}
cisgene <- read.table("data/SNP_50000.txt", stringsAsFactors = F,sep="\t", header=T)
outdfall <- NULL
for (i in 1:dim(gRNAsnp)[1]){
  locsnp <- gRNAsnp[i,"locus_SNP"]
  loc <- gRNAsnp[i,"locus"]
  loccisgene <- unique(cisgene[cisgene[,3]==locsnp,1])
  for (pfile in list.files("data/gRNA_edgeR-QLF/", paste0(loc, "_.*_edgeR-qlf_Neg1_Empricialp.Rd"))){
    load(paste0("data/gRNA_edgeR-QLF/",pfile))
    gRNA <- strsplit(pfile, split = "_edgeR")[[1]][1]
    outdf <- cbind(locsnp, loc, gRNA,loccisgene, resEmpiricalp[loccisgene, c(1,2,4,5,9,10)])
    colnames(outdf)[c(1,2,4,9,10)] <- c("locus_SNP", "locus", "cisGene", "empiricalP", "empiricalFDR")
    outdfall <- rbind(outdfall, outdf)
  }
}
rownames(outdfall) <- NULL
outdfall <- outdfall[complete.cases(outdfall),]
outdfall[, 5:10] <- signif(outdfall[,5:10],3)
write.table( outdfall , file= "data/SNP_50000_empiricalP.txt" , row.names=F, col.names=T, sep="\t", quote = F)
DT::datatable(outdfall)
```
[download this table](https://github.com/simingz/cropseq/blob/master/data/SNP_50000_empiricalP.txt)

## genes +/- 200kb of targeted SNPs
```{r, er200k}
cisgene <- read.table("data/SNP_200000.txt", stringsAsFactors = F,sep="\t", header=T)
outdfall <- NULL
for (i in 1:dim(gRNAsnp)[1]){
  locsnp <- gRNAsnp[i,"locus_SNP"]
  loc <- gRNAsnp[i,"locus"]
  loccisgene <- unique(cisgene[cisgene[,3]==locsnp,1])
  for (pfile in list.files("data/gRNA_edgeR-QLF/", paste0(loc, "_.*_edgeR-qlf_Neg1_Empricialp.Rd"))){
    load(paste0("data/gRNA_edgeR-QLF/",pfile))
    gRNA <- strsplit(pfile, split = "_edgeR")[[1]][1]
    outdf <- cbind(locsnp, loc, gRNA,loccisgene, resEmpiricalp[loccisgene, c(1,2,4,5,9,10)])
    colnames(outdf)[c(1,2,4,9,10)] <- c("locus_SNP", "locus", "cisGene", "empiricalP", "empiricalFDR")
    outdfall <- rbind(outdfall, outdf)
  }
}
rownames(outdfall) <- NULL
outdfall <- outdfall[complete.cases(outdfall),]
outdfall[, 5:10] <- signif(outdfall[,5:10],3)
write.table( outdfall , file= "data/SNP_200000_empiricalP.txt" , row.names=F, col.names=T, sep="\t", quote = F)
DT::datatable(outdfall)
```
[download this table](https://github.com/simingz/cropseq/blob/master/data/SNP_200000_empiricalP.txt)

## genes +/- 500kb of targeted SNPs
```{r, er500k}
cisgene <- read.table("data/SNP_500000.txt", stringsAsFactors = F,sep="\t", header=T)
outdfall <- NULL
for (i in 1:dim(gRNAsnp)[1]){
  locsnp <- gRNAsnp[i,"locus_SNP"]
  loc <- gRNAsnp[i,"locus"]
  loccisgene <- unique(cisgene[cisgene[,3]==locsnp,1])
  for (pfile in list.files("data/gRNA_edgeR-QLF/", paste0(loc, "_.*_edgeR-qlf_Neg1_Empricialp.Rd"))){
    load(paste0("data/gRNA_edgeR-QLF/",pfile))
    gRNA <- strsplit(pfile, split = "_edgeR")[[1]][1]
    outdf <- cbind(locsnp, loc, gRNA,loccisgene, resEmpiricalp[loccisgene, c(1,2,4,5,9,10)])
    colnames(outdf)[c(1,2,4,9,10)] <- c("locus_SNP", "locus", "cisGene", "empiricalP", "empiricalFDR")
    outdfall <- rbind(outdfall, outdf)
  }
}
rownames(outdfall) <- NULL
outdfall <- outdfall[complete.cases(outdfall),]
outdfall[, 5:10] <- signif(outdfall[,5:10],3)
write.table( outdfall , file= "data/SNP_500000_empiricalP.txt" , row.names=F, col.names=T, sep="\t", quote = F)
DT::datatable(outdfall)
```
[download this table](https://github.com/simingz/cropseq/blob/master/data/SNP_500000_empiricalP.txt)
