---
title: "Quality_metrics"
author: "Siming Zhao"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Number of guide RNAs per cell

* number of cells with guide RNA reads =1
From Siwei's cellranger run:
```{r siweimatrix, eval=T}
library(Matrix)
matrix_dir = "/Volumes/CROP-seq/data_from_Siwei/Xin_scRNA_seq_05Nov2018/filtered_gene_bc_matrices/CellRanger_index/"
matrix.path <- paste0(matrix_dir, "matrix.mtx")
dm <- readMM(file = matrix.path)
dm1 <- tail(dm,n=76)
length(colSums(dm1)[colSums(dm1)==1])
```

From Alan's cellranger run:
```{r Alanmatrix, eval=T, cache=T}
matrix_dir1 = "/Volumes/CROP-seq/NSC0507_cellranger/outs/filtered_gene_bc_matrices/cellranger_ref/"
matrix.path1 <- paste0(matrix_dir1, "matrix.mtx")
mattemp1 <- readMM(file = matrix.path1)
mattemp11 <- tail(mattemp1,n=76)
length(colSums(mattemp11)[colSums(mattemp11)==1])

matrix_dir2 = "/Volumes/CROP-seq/NSC08_cellranger/outs/filtered_gene_bc_matrices/cellranger_ref/"
matrix.path2 <- paste0(matrix_dir2, "matrix.mtx")
mattemp2 <- readMM(file = matrix.path2)
mattemp21 <- tail(mattemp2,n=76)
length(colSums(mattemp21)[colSums(mattemp21)==1])
```
Note: in Alan's original analysis conversion from h5 to csv step didn't seem to work properly. if starting from matrix.mtx files. Siwei and Alan's analyses gave the same results. So from now on, we will always start from Siwei's matrix.mtx file. 

* distribution of gRNA types per cell
```{r gRNAdist, eval=T}
barcode.path <- paste0(matrix_dir, "barcodes.tsv")
features.path <- paste0(matrix_dir, "genes.tsv")
feature.names = read.delim(features.path, header = FALSE,
                           stringsAsFactors = FALSE)
barcode.names = read.delim(barcode.path, header = FALSE,
                           stringsAsFactors = FALSE)
colnames(dm) = barcode.names$V1
rownames(dm) = feature.names$V1
dm1 <- tail(dm,n=76)

hist(apply(dm1, 2, function(x) length(x[x>0])),breaks=300,xlim=c(0,15),ylim=c(0,2500), main="Distribution of number of gRNA types per cell", xlab= "# gRNA type per cell")
```

## number of cells targeted for each locus

```{r celldist, eval=T,results='hide',message=F}
library(dplyr)
dm1df <- as.data.frame(as.matrix(dm1))
dm1df$label = sapply(strsplit(rownames(dm1),split = '_'), function(x){x[1]})
dm1dfagg = as.data.frame(dm1df %>% group_by(label) %>% summarise_all(funs(sum)))
row.names(dm1dfagg) =dm1dfagg$label
dm1dfagg$label =NULL
```

* number of cells targeted for each locus
```{r celldist1, eval=T}
ncell <- apply(dm1dfagg,1, function (x) length(x[x>=1]))
barplot(ncell,las=2,cex.lab=1, main= "# cells targted for each locus")
```

* number of cells only targeted for that locus
```{r eval=T}
# Singletons (cells with only 1 gRNA)
nlocus <- apply(dm1dfagg, 2, function (x) length(x[x>=1]))
hist(nlocus,breaks=100, main="number of targeted locus each cell")
dm1dfagg.uni= dm1dfagg[,nlocus==1]

ncell.uni <- apply(dm1dfagg.uni,1, function (x) length(x[x>=1]))
barplot(ncell.uni,las=2,cex.lab=1,main= "# cells uniquely targted for each locus")
```

## UMI count distribution for cells with unique targeted locus
```{r UMI, eval=T}
# Singletons (cells with only 1 targeted locus)
dm.uni <- dm[,nlocus==1]
nUMI <- colSums(dm.uni)
hist(nUMI,breaks=100,xlim=c(0,1e5))  
```

## UMI count distribution for gRNAs in cells with unique targeted locus
```{r UMIgRNA, eval=T}
# Singletons (cells with only 1 targeted locus)
nUMIgRNA <- colSums(tail(dm.uni,76))
hist(nUMIgRNA,breaks=500,xlim=c(0,20), main = "Histogram of nUMI for gRNAs")  
```

## Prepare data for differential gene expression
```{r save, eval=T}
save(dm,dm1dfagg,nlocus, file="data/DE_input.Rd")
```

Parameters used:

* for a cell to be considered targeted uniquely at a locus: total read counts for the 3 gRNAs targeting that locus >1, total read counts for gRNA of other locus=0. 

* negative control: neg_EGFP and neg_CTRL are pooled together.

* cells to be exluded due to low total UMI count: no filtering


